snuba/query/processors/performance_expressions.py

```diff
@@ -1,40 +1,11 @@
-from typing import Optional
-
-from snuba.query.conditions import (
-    binary_condition,
-    BooleanFunctions,
-    ConditionFunctions,
-)
-from snuba.query.dsl import count, countIf, divide, multiply, plus
-from snuba.query.expressions import (
-    Expression,
-    Literal,
-    Column,
-)
+from snuba.clickhouse.columns import ColumnSet, String, UInt
+from snuba.query.validation.signature import Column as ColType
+from snuba.query.processors.custom_function import CustomFunction
 
 
-def apdex(alias: Optional[str], column: Column, satisfied: Literal) -> Expression:
-    tolerated = multiply(satisfied, Literal(None, 4))
-
-    return divide(
-        plus(
-            countIf(binary_condition(None, ConditionFunctions.LTE, column, satisfied)),
-            divide(
-                countIf(
-                    binary_condition(
-                        None,
-                        BooleanFunctions.AND,
-                        binary_condition(
-                            None, ConditionFunctions.GT, column, satisfied,
-                        ),
-                        binary_condition(
-                            None, ConditionFunctions.LTE, column, tolerated,
-                        ),
-                    ),
-                ),
-                Literal(None, 2),
-            ),
-        ),
-        count(),
-        alias,
-    )
+apdex_processor = CustomFunction(
+    ColumnSet([("column", String()), ("satisfied", UInt(32))]),
```

Comment:
you need to pass the abstract Columnset from the dataset you are using here, You cannot define your own. It will be used for validation (you will get it from the dataset method `get_abstract_column_set`).
Also when you will do that you will notice that this does not validate properly because of. `("satisfied", ColType({UInt}))` I assume you want to pass a literal there .... but .... well this is embarrassing, I forgot to implement a validator for literals. So either you implement one or use `Any` as validator.

Reply:
OK that makes sense. I'll take a look at the validation code.